### PR TITLE
plugins/compiler: init

### PR DESCRIPTION
**Pull request by khaneliman**
Resolves https://github.com/nix-community/nixvim/issues/1527

From what I could tell, there's not much to setup with this?

### CODE REVIEW

**Comment by traxys** [comment]
Maybe we should add [this](https://github.com/Zeioth/compiler.nvim?tab=readme-ov-file#required-system-dependencies) warning to our documentation?
> Some languages require you manually install their compilers in your machine, so compiler.nvim is able to call them. [Please check here](https://github.com/Zeioth/Compiler.nvim/wiki/how-to-install-the-required-dependencies), as the packages will be different depending your operative system.

**Comment by khaneliman** [comment]
> Maybe we should add [this](https://github.com/Zeioth/compiler.nvim?tab=readme-ov-file#required-system-dependencies) warning to our documentation?
> 
> > Some languages require you manually install their compilers in your machine, so compiler.nvim is able to call them. [Please check here](https://github.com/Zeioth/Compiler.nvim/wiki/how-to-install-the-required-dependencies), as the packages will be different depending your operative system.

Tried adding the note, is that how you guys will be using the new admonitions?

**Comment by khaneliman** [comment]
Alright, fixed the admonition. 
image

File: plugins/by-name/compiler/default.nix [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  lib,
+  ...
+}:
+lib.nixvim.neovim-plugin.mkNeovimPlugin {
+  name = "compiler";
+  originalName = "compiler.nvim";
+  package = "compiler-nvim";
```
Comment (MattSturgeon) [inline plugins/by-name/compiler/default.nix:8]: Should we be passing `callSetup = false` and/or `hasSettings = false`?

Comment (khaneliman) [reply plugins/by-name/compiler/default.nix:8]: It looked like upstream had a setup function and passes an opts table, but they don't currently have documented configuration.

Comment (MattSturgeon) [reply plugins/by-name/compiler/default.nix:8]: The lazy.nvim usage example includes `opts = {}`, making me think they _do_ have a setup function and maybe a settings table?

Comment (khaneliman) [reply plugins/by-name/compiler/default.nix:8]: So, I didn't want to restrict future configuration if they do start using the opts that can be provided. https://github.com/Zeioth/compiler.nvim/blob/e889774c7f2fefad60f8e6638d324223b07d2624/lua/compiler/init.lua#L9

Comment (MattSturgeon) [reply plugins/by-name/compiler/default.nix:8]: I came to the same conclusion when I did my own research :+1: 

The setup function does take an opts parameter, but it is currently unused internally.

I agree we should keep things as-is.

**Comment by khaneliman** [comment]
@mergifyio queue

**Pull request merged**